src/main/java/com/listywave/list/application/service/CommentService.java

```diff
@@ -0,0 +1,75 @@
+package com.listywave.list.application.service;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+import com.listywave.auth.application.domain.JwtManager;
+import com.listywave.list.application.domain.Comment;
+import com.listywave.list.application.domain.Lists;
+import com.listywave.list.application.domain.Reply;
+import com.listywave.list.application.dto.response.CommentCreateResponse;
+import com.listywave.list.application.dto.response.CommentFindResponse;
+import com.listywave.list.repository.CommentRepository;
+import com.listywave.list.repository.ListRepository;
+import com.listywave.list.repository.ReplyRepository;
+import com.listywave.user.application.domain.User;
+import com.listywave.user.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class CommentService {
+
+    private final JwtManager jwtManager;
+    private final ListRepository listRepository;
+    private final UserRepository userRepository;
+    private final ReplyRepository replyRepository;
+    private final CommentRepository commentRepository;
+
+    public CommentCreateResponse create(Long listId, String accessToken, String content) {
+        Long userId = jwtManager.read(accessToken);
+        User user = userRepository.getById(userId);
+        Lists list = listRepository.getById(listId);
+
+        Comment comment = Comment.create(list, user, content);
+        Comment saved = commentRepository.save(comment);
+
+        return CommentCreateResponse.of(saved, user);
+    }
+
+    public CommentFindResponse getComments(Long listId, int size, Long cursorId) {
+        Lists list = listRepository.getById(listId);
+        Long totalCount = commentRepository.countByList(list);
+
+        List<Comment> comments = commentRepository.getComments(listId, size, cursorId);
+        Map<Comment, List<Reply>> result = comments.stream()
+                .collect(toMap(
+                        identity(),
+                        replyRepository::getAllByComment
+                ));
+
+        Long cursorIdOfResult = comments.get(comments.size() - 1).getId();
+        boolean hasNext = comments.size() >= size;
```

Comment:
여기서 댓글이 총 10개이고 size를 5로 5개씩 가져온 경우 두번째 떄는 다음페이지가 없을텐데
해당 코드를 보면 `boolean hasNext = comments.size() >= size;` 는 2번째 페이지에서도 다음페이지가 있다고 판단 될 거 같네요

그리고 만약 아래 제안 드린 cursorId 방식을 사용한다면 comments.stream() 들어가기전에 hasNext 체크하고 true이면 comments의 끝번째 인덱스를 remove시키고 toMap으로 변경하는 게 좋을 거 같네요

```suggestion
        Long cursorIdOfResult = comments.get(comments.size() - 1).getId();
        boolean hasNext = comments.size() >size;
```